packages/account-sdk/src/interface/public-utilities/spend-permission/utils.ts

```diff
@@ -0,0 +1,128 @@
+import { spendPermissionManagerAddress } from ':sign/base-account/utils/constants.js';
+import { Address, Hex, getAddress } from 'viem';
+import { RequestSpendPermissionType } from './methods/requestSpendPermission.js';
+
+const SPEND_PERMISSION_TYPED_DATA_TYPES = {
+  SpendPermission: [
+    {
+      name: 'account',
+      type: 'address',
+    },
+    {
+      name: 'spender',
+      type: 'address',
+    },
+    {
+      name: 'token',
+      type: 'address',
+    },
+    {
+      name: 'allowance',
+      type: 'uint160',
+    },
+    {
+      name: 'period',
+      type: 'uint48',
+    },
+    {
+      name: 'start',
+      type: 'uint48',
+    },
+    {
+      name: 'end',
+      type: 'uint48',
+    },
+    {
+      name: 'salt',
+      type: 'uint256',
+    },
+    {
+      name: 'extraData',
+      type: 'bytes',
+    },
+  ],
+};
+
+export type SpendPermissionTypedData = {
+  domain: {
+    name: 'Spend Permission Manager';
+    version: '1';
+    chainId: number;
+    verifyingContract: typeof spendPermissionManagerAddress;
+  };
+  types: typeof SPEND_PERMISSION_TYPED_DATA_TYPES;
+  primaryType: 'SpendPermission';
+  message: {
+    account: Address;
+    spender: Address;
+    token: Address;
+    allowance: string;
+    period: number;
+    start: number;
+    end: number;
+    salt: string;
+    extraData: Hex;
+  };
+};
+
+export function createSpendPermissionTypedData(
+  request: RequestSpendPermissionType
+): SpendPermissionTypedData {
+  const { account, spender, token, chainId, allowance, periodInDays, start, end, salt, extraData } =
+    request;
+
+  return {
+    domain: {
+      name: 'Spend Permission Manager',
+      version: '1',
+      chainId: chainId,
+      verifyingContract: spendPermissionManagerAddress,
+    },
+    types: SPEND_PERMISSION_TYPED_DATA_TYPES,
+    primaryType: 'SpendPermission',
+    message: {
+      account: getAddress(account),
+      spender: getAddress(spender),
+      token: getAddress(token),
+      allowance: allowance.toString(),
+      period: 86400 * periodInDays,
+      start: toTimestampInSeconds(start ?? new Date()),
+      end: toTimestampInSeconds(end ?? add100Years(new Date())),
+      salt: salt ?? getRandomHexString(32),
```

Comment:
@spencerstock wanna have you double check this logic, salt will always be 32 byte here

Reply:
Looks good to me!